### PR TITLE
add titan network

### PR DESCRIPTION
**Pull request by nezha90**

### CODE REVIEW

**Comment by JeremyParish69** [comment]
This looks to be a good submission. The only issue I notice is that the 'chain' 'ttnt' images seem to be duplicates, which I believe can be reduced to a single set of images.

**Comment by nezha90** [comment]
> This looks to be a good submission. The only issue I notice is that the 'chain' 'ttnt' images seem to be duplicates, which I believe can be reduced to a single set of images.

Thanks, I changed this

File: titannet/chain.json [hunk 1]
```diff
@@ -0,0 +1,126 @@
+{
+  "$schema": "../chain.schema.json",
+  "chain_name": "titannet",
+  "chain_type": "cosmos",
+  "chain_id": "titan-test-4",
```
Comment (JeremyParish69) [inline titannet/chain.json:5]: Is this a testnet?

File: _IBC/osmosis-titan.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  "$schema": "../ibc_data.schema.json",
+  "chain_1": {
+    "chain_name": "osmosis",
+    "client_id": "07-tendermint-3383",
+    "connection_id": "connection-2838"
+  },
+  "chain_2": {
+    "chain_name": "titannet",
```
Comment (JeremyParish69) [inline _IBC/osmosis-titan.json:9]: This file name must match the chain_names, so it must be `osmosis-titannet.json`

Comment (nezha90) [reply _IBC/osmosis-titan.json]: > This file name must match the chain_names, so it must be `osmosis-titannet.json`

Thank you, modified

File: titannet/chain.json (hunk 1 above)
Comment (nezha90) [reply titannet/chain.json:5]: > Is this a testnet?

Yes, this is indeed a testnet (titan-test-4) as shown in the chain.json configuration.
We're in the final preparation phase before mainnet launch. At this stage, we aim to test IBC channel connectivity between our Testnet and OSMO, particularly focusing on TNT4 cross-chain transfers to OSMO to validate additional functionalities.
Both our team and community are fully aware of its testnet status. We kindly request approval for this submission. We're open to any modification requirements or guidelines you may have and can adjust according to your specifications. What are your requirements for this integration?

Comment (JeremyParish69) [reply titannet/chain.json:5]: In this case, the value for `"network_type"` shouldn't be `"mainnet"`, but instead should be `"testnet"`.
In addition, the chain_name should be titannettestnet (in the assetlist.json, the chain.json, versions.json, all ibc files, and the directory name. The directory must also be relocated inside the /testnets/ directory, where you'll see that all other testnets have chain_name as <mainnet name>+'testnet'.

File: titannet/chain.json [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  "website": "https://titannet.io/",
+  "pretty_name": "Titan Network",
+  "status": "live",
+  "network_type": "mainnet",
```
Comment (JeremyParish69) [inline titannet/chain.json:9]: Not a mainnet

File: titannet/assetlist.json [hunk 4]
```diff
@@ -0,0 +1,78 @@
+{
+  "$schema": "../assetlist.schema.json",
+  "chain_name": "titannet",
```
Comment (JeremyParish69) [inline titannet/assetlist.json:3]: "titannettestnet"

File: _IBC/osmosis-titannet.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
+{
+  "$schema": "../ibc_data.schema.json",
+  "chain_1": {
+    "chain_name": "osmosis",
+    "client_id": "07-tendermint-3383",
+    "connection_id": "connection-2838"
+  },
+  "chain_2": {
+    "chain_name": "titannet",
```
Comment (JeremyParish69) [inline _IBC/osmosis-titannet.json:9]: "titannettestnet"

File: titannet/chain.json [hunk 6]
```diff
@@ -0,0 +1,126 @@
+{
+  "$schema": "../chain.schema.json",
+  "chain_name": "titannet",
```
Comment (JeremyParish69) [inline titannet/chain.json:3]: "titannettestnet"

File: titannet/versions.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{
+  "$schema": "../versions.schema.json",
+  "chain_name": "titannet",
```
Comment (JeremyParish69) [inline titannet/versions.json:3]: "titannettestnet"

File: titannet/chain.json [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+      "tx_page": "https://testnet.titan.explorers.guru/transaction/${txHash}",
+      "account_page": "https://testnet.titan.explorers.guru/account/${accountAddress}",
+      "validator_page": "https://testnet.titan.explorers.guru/validator/${validatorAddress}",
+      "proposal_page": "https://testnet.titan.explorers.guru/proposal/${proposalId}",
+      "block_page": "https://testnet.titan.explorers.guru/block/${blockHeight}"
+    }
+  ],
+  "images": [
+    {
+      "png": "https://raw.githubusercontent.com/cosmos/chain-registry/master/titanchain/images/ttnt.png",
```
Comment (JeremyParish69) [inline titannet/chain.json:119]: if the images are relocated to /testnets/titannettestnet/images/, then all these path must also be updated. For now, it is permissible to leave the images under /titannet/images/, but many teams add them images to the testnet images directory first, and then move them over to the mainnet one once the mainnet is registered.

**Pull request closed without merging**